tests/e2e/linalg_ext_ops/pack.mlir

```diff
@@ -7,6 +7,27 @@ func.func @pack_simple() {
   return
 }
 
+func.func @dynamic_pack_simple() {
+  %iree_input = flow.tensor.constant dense<[
```

Comment:
Instead of hard coding the inputs could we use the `generic` to create a source input and verify correctness for pack as being same as `pad + reshape + transpose` like it is done for the other cases. That way you can add a "large" test which will make sure this works in parallel execution as well (small ops might be made sequential, it isnt today).

Reply:
I'm not sure how to ask compiler not being too smart for this case. `flow.tensor.constant` is a guard to prevent folding, which let the backend get dynamic shapes as input. I can do some study later.

Reply:
You can use `util.unfoldable.constant`?

Reply:
`flow.tensor.constant` is the dynamic version of `util.unfoldable.constant`. We can't use `util.unfoldable.constant` to create dynamic inputs because canonicalizer would fold it away. (E.g., if we create a tensor.cast op which cast the result into dynamic case, the tensor.cast op will just be folded away.)

We probably need `flow.scalar.constant` in this case, but it doesn't exist today.

Reply:
Ok, I see your point. What I was thinking of was something like this
```
%d0 = util.unfoldable.constant <some index value>
%d1 = util.unfoldable.constant <some index value>
%0 = linalg.init_tensor [%d0, %d1]  : tensor<?x?xf32>
%source = linalg.generic outs(%0 :  tensor<?x?xf32>) { ... } // see "large" tests above for the way to initialize the source
%pack = iree_linalg_ext.pack %source....
%high_pad0 = /// some function of %d0
%high_pad1 = /// some function of %d1
%pad = tensor.pad %source low [0, 0] high [%high_pad0, %high_pad1]
%reshape = tensor.expand_shape %pad
%transpose = linalg.generic {...} // same as "large" tests above
check.eq(%transpose, %pack).
```

basically make the  `%d0` and `%d1` unfoldable constants...

Reply:
I see your point. Just realized that `util.unfoldable.constant <some index value>` is what I'm looking for (i.e., `flow.scalar.constant`). I'll give it a shot.

Reply:
it works, thanks for the help!